tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -12628,11 +12629,30 @@ gen_acon_expr(int ilix, LL_Type *expected_type)
   sym_is_refd(sptr);
   idx = (STYPEG(sptr) == ST_STRUCT || STYPEG(sptr) == ST_ARRAY
          || ACONOFFG(opnd) < 0) ? 0 : ACONOFFG(opnd);
-  process_sptr_offset(sptr, variable_offset_in_aggregate(sptr, idx));
+  if (DTA(dtype)) {
+    /* handle the flang1's pragma !dir$ align
+     * adjust the sptr's offset in ag to be aligned with
+     * sptr's alignment.
+     * */
+    process_sptr_offset(sptr, idx + ALIGN(
+                                      variable_offset_in_aggregate(sptr, 0),
+                                      DTA(dtype)));
+  }else {
+    process_sptr_offset(sptr, variable_offset_in_aggregate(sptr, idx));
+  }
   idx = ACONOFFG(opnd); /* byte offset */
 
   ty1 = make_lltype_from_dtype(DT_ADDR);
-  idx = variable_offset_in_aggregate(sptr, idx);
+  if (DTA(dtype)) {
```

Comment:
Ditto.

Reply:
> Ditto.

I have formatted the statements using clang-format.